protocol/0092-SAMM-spot_automated_market_maker.md

```diff
@@ -0,0 +1,223 @@
+
+$$
+L = \frac{c_q}{\sqrt{p_u} - \sqrt{p_l}} ,
+$$
+
+$$
+c_b = 0 ,
+$$
+
+
+where $c_q$ is the quote commitment value specified (note that as $p_r >= p_l$ it would be invalid to specify a base commitment value).
+
+In the final case where $p_l < p_r < p_u$ we can think of there being two separate ranges, one above (from $p_r$ to $p_u$) and one below ($p_l$ to $p_r$). In the upper range the AMM is fully in the base asset, in the lower it is fully in the quote asset. Thus, the protocol can take the specified commitment value, calculate $L$ with that on the relevant range.
+
+Concretely:
+
+If $c_q$ is specified
+
+$$
+L = \frac{c_q}{\sqrt{p_r} - \sqrt{p_l}} ,
+$$
+
+$$
+c_b = L \frac{\sqrt{p_u} - \sqrt{p_r}}{\sqrt{p_u} \sqrt{p_r}}
+$$
+
+and if $c_b$ is specified
+
+$$
+L = c_b \frac{\sqrt{p_l} \sqrt{p_r}}{\sqrt{p_r} - \sqrt{p_l}} ,
+$$
+
+$$
+c_q = L (\sqrt{p_r} - \sqrt{p_l}) ,
+$$
+
+#### Fair price
+
+The fair price can then be calculated as
+
+$$
+p_f = \frac{b_q + L \sqrt{p_l}}{b_b + \frac{L}{\sqrt{p_u}}} ,
+$$
+
+where $b_q$ is the current balance of the quote token and $b_b$ is the current balance of the base token.
+
+#### Price to trade a given volume
+
+Finally, the protocol needs to calculate the inverse of the previous section. That is, given a volume bought from/sold to the AMM, at what price should the trade be executed. This could be calculated naively by summing across all the smallest increment volume differences, however this would be computationally inefficient and can be optimised by instead considering the full trade size.
+
+First, the virtual `x` and `y` balances must be found (where `x` is the base balance and `y` is the quote balance):
+
+$$
+x_v = b_b + \frac{L}{\sqrt{p_u}} ,
+$$
+
+$$
+y_v = b_q + L \sqrt{p_l} ,
+$$
+
+Once obtained, the price can be obtained from the fundamental requirement of the product $y \cdot x$ remaining constant. This gives the relationship
+
+$$
+y_v \cdot x_v = (y_v + \Delta y) \cdot (x_v - \Delta x) ,
+$$
+
+From which $\Delta y$ must be calculated
+
+$$
+\Delta y = \frac{y_v \cdot x_v}{x_v - \Delta x} - y_v ,
+$$
+
+Thus giving a final execution price to return of $\frac{\Delta y}{\Delta x}$.
+
+#### Volume between two prices
+
+For the second interface one needs to calculate the volume which would be posted to the book between two price levels. In order to calculate this for an AMM one is ultimately asking the question "what volume of swap would cause the fair price to move from price A to price B?"
+
+To calculate this, the interface will need the `starting price` $p_s$, `ending price` $p_e$, `upper price` $p_u$ and the `L`. At `P = 0` use the values for the range which the volume change will cause the position to move into.
+
+First, calculate the implied position at `starting price` and `ending price` and return the difference.
+
+For a given price $p$ calculate implied position $P_i$ with
```

Comment:
Nitpick, but using a different letter for implied position would make it a bit easier to follow, plenty of `p`s going around already.